src/test/java/edu/mit/lib/mama/MamaTest.java

```diff
@@ -0,0 +1,124 @@
+           hdl.execute("insert into metadatafieldregistry (metadata_field_id, metadata_schema_id, element) values(4, 1, 'creator')");
+           hdl.execute("create table item (item_id int primary key)");
+           hdl.execute("insert into item (item_id) values(1)");
+           hdl.execute("insert into item (item_id) values(2)");
+           hdl.execute("create table metadatavalue (metadata_value_id int primary key, item_id int, metadata_field_id int, text_value varchar)");
+           hdl.execute("insert into metadatavalue (metadata_value_id, item_id, metadata_field_id, text_value) values(1, 1, 1, 'http://hdl.handle.net/123456789/3')");
+           hdl.execute("insert into metadatavalue (metadata_value_id, item_id, metadata_field_id, text_value) values(2, 2, 2, 'A Very Important Study')");
+        }
+        // launch api service in it's own thread - then give it time to initialize before firing tests at it
+        apiService.start();
+        Thread.currentThread().sleep(2000);
+    }
+
+    @Test
+    public void apiSpecValidity() throws IOException {
+        assertThat(api.validate(), validates());
+    }
+
+    @Test
+    public void basicRequest() throws IOException {
+        String url = TEST_SVC_URL + "/item?qf=dc.identifier.uri&qv=http://hdl.handle.net/123456789/3";
+        HttpResponse response = baseClient.execute(new HttpGet(url));
+        assertThat(baseClient.getLastReport(), checks());
+    }
+
+    @Test
+    public void usageSuite() throws IOException {
+        final SimpleReportAggregator aggregator = new SimpleReportAggregator();
+        final RamlHttpClient client = baseClient.aggregating(aggregator);
+
+        // URLs must collectively trigger full set of API behaviors as described by spec
+        // missing required query parameter 'qf' - should return 400 response code
+        send(client, TEST_SVC_URL + "/item?qv=http://hdl.handle.net/123456789/3");
+        // required query parameter 'qf' lacking value - should return 400 response code
+        send(client, TEST_SVC_URL + "/item?qf=&qv=http://hdl.handle.net/123456789/3");
+        // required query parameter 'qf' bogus value - should return 404 response code
+        send(client, TEST_SVC_URL + "/item?qf=dc.identifier.foo&qv=http://hdl.handle.net/123456789/3");
+        // no items matching query - should return 404 response code
+        send(client, TEST_SVC_URL + "/item?qf=dc.identifier.uri&qv=http://hdl.handle.net/123456789/5");
+        // correct query - no response parameters
+        send(client, TEST_SVC_URL + "/item?qf=dc.identifier.uri&qv=http://hdl.handle.net/123456789/3");
+        // correct query - optional response parameters
+        send(client, TEST_SVC_URL + "/item?qf=dc.identifier.uri&qv=http://hdl.handle.net/123456789/3&rf=dc.title&rf=dc.type");
+
+        assertUsage(aggregator.getUsage(), EnumSet.allOf(UsageItem.class));
```

Comment:
Yea, that's pretty much it - it adds this dimension to testing where you are forced to all more tests to make it match the spec - otherwise that test will fail